router/transformer/transformer.go

```diff
@@ -183,43 +195,18 @@ func (trans *HandleT) Transform(transformType string, transformMessage *types.Tr
-	trans.logger.Debugf(`[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Proxy Request starts - %[1]v`, destName, jobId)
-	rawJSON, err := jsonfast.Marshal(responseData)
-	if err != nil {
-		panic(err)
-	}
-
-	var respData []byte
-	var respCode int
-
-	url := getProxyURL(destName)
-	payload := rawJSON
-
-	operation := func() error {
-		var requestError error
-		trans.logger.Debugf(`[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Proxy Request operation method - %[1]v`, destName, jobId)
-		// start
-		rdl_time := time.Now()
-		respData, respCode, requestError = trans.makeHTTPRequest(ctx, url, payload, destName, jobId)
-		reqSuccessStr := strconv.FormatBool(requestError == nil)
-		stats.NewTaggedStat("transformer_proxy.request_latency", stats.TimerType, stats.Tags{"requestSuccess": reqSuccessStr, "destination": destName}).SendTiming(time.Since(rdl_time))
-		stats.NewTaggedStat("transformer_proxy.request_result", stats.CountType, stats.Tags{"requestSuccess": reqSuccessStr, "destination": destName}).Increment()
-		trans.logger.Debugf(`[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} RespData - %[3]v, RespCode - %[4]v `, destName, jobId, string(respData), respCode)
-		trans.logger.Debugf(`[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Proxy Request operation ended - %[1]v`, destName, jobId)
-		// end
-		return requestError
-	}
+func (trans *HandleT) ProxyRequest(ctx context.Context, proxyReqParams *ProxyRequestParams) (int, string) {
+	stats.NewTaggedStat("transformer_proxy.delivery_request", stats.CountType, stats.Tags{"destination": proxyReqParams.DestName}).Increment()
+	trans.logger.Debugf(`[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Proxy Request starts - %[1]v`, proxyReqParams.DestName, proxyReqParams.JobId)
 
-	backoffWithMaxRetry := backoff.WithMaxRetries(backoff.NewExponentialBackOff(), uint64(retryWithBackoffCount))
-	err = backoff.RetryNotify(operation, backoffWithMaxRetry, func(err error, t time.Duration) {
-		pkgLogger.Errorf("[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Request for proxy to URL:: %[3]v, Error:: %+[4]v retrying after:: %[5]v,", destName, jobId, url, err, t)
-		stats.NewTaggedStat("transformer_proxy.retries", stats.CountType, stats.Tags{"destination": destName}).Increment()
-	})
+	rdl_time := time.Now()
+	respData, respCode, requestError := trans.makeTfProxyRequest(ctx, proxyReqParams)
+	reqSuccessStr := strconv.FormatBool(requestError == nil)
+	stats.NewTaggedStat("transformer_proxy.request_latency", stats.TimerType, stats.Tags{"requestSuccess": reqSuccessStr, "destination": proxyReqParams.DestName}).SendTiming(time.Since(rdl_time))
+	stats.NewTaggedStat("transformer_proxy.request_result", stats.CountType, stats.Tags{"requestSuccess": reqSuccessStr, "destination": proxyReqParams.DestName}).Increment()
 
-	if err != nil {
-		panic(fmt.Errorf("[TransformerProxy] Proxy request failed after max retries Error:: %+v", err))
+	if requestError != nil {
+		return respCode, requestError.Error()
```

Comment:
`http.DetectContentType(respData)`

Why do we do this by default? Isn't this a costly operation

Reply:
It was changed as part of some kind of review comment or something a long-time ago. Did we see any performance impact due to using `DetectContentType` in general ?